source/server/guarddog_impl.cc

```diff
@@ -61,7 +61,7 @@ GuardDogImpl::GuardDogImpl(Stats::Scope& stats_scope, const Server::Configuratio
 
         // We should be able to share the dispatcher since guard dog's lifetime
         // should eclipse those of actions.
-        Configuration::GuardDogActionFactoryContext context = {api, *dispatcher_};
+        Configuration::GuardDogActionFactoryContext context = {api, *dispatcher_, stats_scope};
```

Comment:
Do you plan to use these stats for the profile action?  You may want to consider making that change now and add some basic test coverage to verify that the other parts of this change work as expected.

Reply:
Done.